can/interfaces/usb2can/serial_selector.py

```diff
@@ -27,14 +29,19 @@ def WMIDateStringToDate(dtmDate):
     return strDateTime
 
 
-def serial():
-    strComputer = '.'
+def find_serial_devices(serial_matcher="ED"):
+    """
+    Finds a list of serial devices.
+
+    :rtype: List[str]
+    """
     objWMIService = win32com.client.Dispatch("WbemScripting.SWbemLocator")
-    objSWbemServices = objWMIService.ConnectServer(strComputer, "root\cimv2")
-    colItems = objSWbemServices.ExecQuery("SELECT * FROM Win32_USBControllerDevice")
-
-    for objItem in colItems:
-        string = objItem.Dependent
-        # find based on beginning of serial
-        if 'ED' in string:
-            return string[len(string) - 9:len(string) - 1]
+    objSWbemServices = objWMIService.ConnectServer(".", "root\cimv2")
+    items = objSWbemServices.ExecQuery("SELECT * FROM Win32_USBControllerDevice")
```

Comment:
Okay, should I simply remove that condition? (@rusoku)

Reply:
Your current solution is already a big improvement, since it searches for the serial number prefix only within the last eight characters, not within the entire entry.

May I suggest another refinement that matches the given (partial) serial number **only at the beginning**?  So "ED" matches "ED123456", but not "FEDCBA98"?